x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -586,8 +590,34 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                   intendedTimestamp,
                 });
 
+                const matchingBuildingBlockAlerts =
+                  newAlerts?.length > 0 && getMatchingBuildingBlockAlerts != null
+                    ? newAlerts.flatMap(
+                        (newAlert) =>
+                          getMatchingBuildingBlockAlerts(newAlert?._source) as Array<{
+                            _id: string;
+                            _source: Record<string, string | number | null>;
+                          }>
+                      )
+                    : [];
+
+                const augmentedBuildingBlockAlerts =
+                  getMatchingBuildingBlockAlerts != null && newAlerts?.length > 0
```

Comment:
double checking if this conditional clause is needed, or perhaps incorrect, since neither ` getMatchingBuildingBlockAlerts` nor `newAlerts` is directly referenced in either case of the conditional